components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java

```diff
@@ -169,4 +169,73 @@ private static Instant parseDateTime(String dateTimeString) {
         }
         return localDateTime;
     }
+
+    /**
+     * Util function to mask content.
+     *
+     * @param content Content that needs to be masked.
+     * @return masked content.
+     */
+    public static String maskContent(String content) {
+
+        if (StringUtils.isNotEmpty(content)) {
+            content = content.replaceAll(LogConstants.CONTENT_MASKING_REGEX, LogConstants.MASKING_CHARACTER);
```

Comment:
replaceAll may have a performance hit. check if we can use a compiled pattern

Reply:
resolved.
[https://stackoverflow.com/questions/6262397/string-replaceall-is-considerably-slower-than-doing-the-job-yourself](https://stackoverflow.com/questions/6262397/string-replaceall-is-considerably-slower-than-doing-the-job-yourself)